repose-aggregator/src/docs/asciidoc/architecture/container.adoc

```diff
@@ -0,0 +1,252 @@
+</repose-container>
+----
+
+== Whitelisting and Blacklisting Ciphers and Protocols
+[WARNING]
+Since security is a constantly moving target, any recommended configuration would quickly become out of date.
+A risk assessment should always be performed by the appropriately qualified people for your organization.
+Links to industry-standard references are provided in the SSL References section below.
+
+Repose supports whitelisting and blacklisting specific protocols and ciphers by exposing portions of the Jetty's configuration via the `container.cfg.xml` file.
+You can use this feature if a specific protocol or cipher has been compromised and you want to block its usage and secure Repose.
+
+In the following example, the container configuration includes TLS v1.2 and TLS ciphers and excludes SSLv3 and SSL protocols.
+
+[source,xml]
+.container.cfg.xml - Protocols and ciphers
+----
+<repose-container xmlns='http://docs.openrepose.org/repose/container/v2.0'>
+    <deployment-config>
+       <deployment-directory auto-clean="false">/var/repose</deployment-directory>
+       <artifact-directory check-interval="60000">/usr/share/repose/filters</artifact-directory>
+       <logging-configuration href="file:///etc/repose/log4j2.xml"/>
+       <ssl-configuration>
+          <keystore-filename>keystore.jks</keystore-filename>
+          <keystore-password>password</keystore-password>
+          <key-password>password</key-password>
+          <included-ciphers>
+              <cipher>.*TLS.*</cipher>
+          </included-ciphers>
+          <excluded-ciphers>
+              <cipher>.*SSL.*</cipher>
+          </excluded-ciphers>
+          <excluded-protocols>
+               <protocol>SSLv3</protocol>
+          </excluded-protocols>
+          <included-protocols>
+              <protocol>TLSv1.2</protocol>
+          </included-protocols>
+          <tls-renegotiation-allowed>false</tls-renegotiation-allowed>
+      </ssl-configuration>
+    </deployment-config>
+</repose-container>
+----
+
+[NOTE]
+You need to specify your keystore in the container configuration just as you would in Jetty.
+
+== Diffie-Hellman Security Risk and Key Size
+Certain attacks (such as Logjam) leverage the weakness of "small" Diffie-Hellman (DH) keys.
+To mitigate the risk of such attackers, users may either exclude vulnerable ciphers, or lengthen the DH keys used by Repose.
+Instructions for the former are above.
+For the latter, note the following:
+
+"Diffie-Hellman (DH) keys of sizes less than 1024 bits have been deprecated because of their insufficient strength.
+In JDK 8, you can customize the ephemeral DH key size with the system property jdk.tls.ephemeralDHKeySize."
+
+In other words, the Java option "-Djdk.tls.ephemeralDHKeySize=2048" can be passed when starting Repose to force the use of longer DH keys.
+
+For more details, see:
+* https://docs.oracle.com/javase/8/docs/technotes/guides/security/jsse/JSSERefGuide.html#customizing_dh_keys
+
+== SSL/TLS References
```

Comment:
I kind of think all of these SSL sections should be under one big SSL section maybe?